patches/aspnetcore/aspnetcore-arm64.patch

```diff
@@ -0,0 +1,109 @@
+diff --git a/Directory.Build.props b/Directory.Build.props
+index 2ef491a409..8e699a2d93 100644
+--- a/Directory.Build.props
++++ b/Directory.Build.props
+@@ -105,6 +105,8 @@
+     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('Windows'))">win</TargetOsName>
+     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('OSX'))">osx</TargetOsName>
+     <TargetOsName Condition=" '$(TargetOsName)' == '' AND $([MSBuild]::IsOSPlatform('Linux'))">linux</TargetOsName>
++    <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant())</BuildArchitecture>
++    <TargetArchitecture Condition="'$(TargetArchitecture)' == '' AND '$(BuildArchitecture)' == 'arm64'">$(BuildArchitecture)</TargetArchitecture>
+     <TargetArchitecture Condition="'$(TargetArchitecture)' == ''">x64</TargetArchitecture>
+     <TargetRuntimeIdentifier>$(TargetOsName)-$(TargetArchitecture)</TargetRuntimeIdentifier>
+ 
+diff --git a/build.sh b/build.sh
+index 05e1628368..bbb79aea08 100755
+--- a/build.sh
++++ b/build.sh
+@@ -13,7 +13,7 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ target_os_name=''
+ ci=false
+ use_default_binary_log=false
+-verbosity='minimal'
++verbosity='detailed'
+ run_restore=''
+ run_build=true
+ run_pack=false
+@@ -27,7 +27,7 @@ build_nodejs=''
+ build_java=''
+ build_installers=''
+ build_projects=''
+-target_arch='x64'
++target_arch=''
+ configuration=''
+ 
+ if [ "$(uname)" = "Darwin" ]; then
+@@ -96,6 +96,21 @@ __warn() {
+ # main
+ #
+ 
++case "$(uname -m)" in
++    aarch64)
++        target_arch='arm64'
++        ;;
++    armv7l)
++        target_arch='arm'
++        ;;
++    i686)
++        target_arch='x86'
++        ;;
++    x86_64)
++        target_arch='x64'
++        ;;
++esac
++
+ while [[ $# -gt 0 ]]; do
+     opt="$(echo "${1/#--/-}" | awk '{print tolower($0)}')"
+     case "$opt" in
+diff --git a/global.json b/global.json
+index b0204615a6..ddc97517fe 100644
+--- a/global.json
++++ b/global.json
+@@ -5,10 +5,7 @@
+   "tools": {
+     "dotnet": "3.0.100-preview8-013656",
+     "runtimes": {
+-      "dotnet/x64": [
+-        "$(MicrosoftNETCoreAppRuntimeVersion)"
+-      ],
+-      "dotnet/x86": [
++      "dotnet": [
+         "$(MicrosoftNETCoreAppRuntimeVersion)"
+       ]
+     },
+diff --git a/src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj b/src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj
```

Comment:
A newer version of this is here: https://github.com/aspnet/AspNetCore/pull/15354/